instance/tests/models/test_instance.py

```diff
@@ -354,23 +354,20 @@ def test_run_playbook_logging(self, mock_run_playbook):
         """
         Ensure logging routines are working on _run_playbook method
         """
-        with tempfile.NamedTemporaryFile() as stdout:
-            with tempfile.NamedTemporaryFile() as stderr:
-
-                mock_run_playbook.return_value.__enter__.return_value.stdout = open(stdout.name, "rb")
-                mock_run_playbook.return_value.__enter__.return_value.stderr = open(stderr.name, "rb")
-                mock_run_playbook.return_value.__enter__.return_value.returncode = 0
-
-                stdout.write(b"HELLO\n")
-                stdout.flush()
-                stderr.write(b"HI\n")
-                stderr.flush()
-
-                instance = OpenEdXInstanceFactory()
-                log, returncode = instance._run_playbook("requirements", "playbook")
-
-                self.assertCountEqual(log, ['HELLO', 'HI'])
-                self.assertEqual(returncode, 0)
+        stdout_r, stdout_w = os.pipe()
+        stderr_r, stderr_w = os.pipe()
+        with open(stdout_r, 'rb', buffering=0) as stdout, open(stderr_r, 'rb', buffering=0) as stderr:
+            mock_run_playbook.return_value.__enter__.return_value.stdout = stdout
+            mock_run_playbook.return_value.__enter__.return_value.stderr = stderr
+            mock_run_playbook.return_value.__enter__.return_value.returncode = 0
+            os.write(stdout_w, b'Hello\n')
+            os.close(stdout_w)
+            os.write(stderr_w, b'Hi\n')
+            os.close(stderr_w)
+            instance = OpenEdXInstanceFactory()
+            log, returncode = instance._run_playbook("requirements", "playbook")
+            self.assertCountEqual(log, ['Hello', 'Hi'])
+            self.assertEqual(returncode, 0)
```

Comment:
@mtyaka To test the timeouts as used in `poll_streams()`, I would have to mock out the selector, which is cumbersome and doesn't provide enough benefit in my opinion.  As a basic level of testing, I added a unit test for the helper function that yields the timeouts instead.